examples/tests/external_logging_stream.py

```diff
@@ -0,0 +1,23 @@
+import logging
+
+import matplotlib.pyplot as plt
+
+from avocado import Test
+
+
+class MatplotlibTest(Test):
+    def test(self):
+
+        logging.getLogger("matplotlib").level = logging.DEBUG
```

Comment:
Hmmm but something like this doesn't work on my side and I need this to make it work:
```python
log.getLogger("matplotlib").parent = log.getLogger("avocado.test")
```
In this way the `matplotlib` logger is told to delegate its messages to the test namespace and the debug.log. Is this correct or am I missing something? On my side if I don't use the above line the output will appear in the [stderr] stream even with the most up-to-date changes here so far.

Reply:
The line 11 of this example:
```
        logging.getLogger("matplotlib").level = logging.DEBUG
```
only enables logging in `matplotlib` and it is not necessary for avocado, if your library already generating logs you don't need to do any changes. But what you have to do is to tell avocado to store those logs by `--store-logging-stream=matplotlib` option

> Hmmm but something like this doesn't work on my side and I need this to make it work:
> 
> ```python
> logging.getLogger("matplotlib").parent = logging.getLogger("avocado.test")
> ```
> 
> In this way the `matplotlib` logger is told to delegate its messages to the test namespace and the debug.log. Is this correct or am I missing something? On my side if I don't use the above line the output will appear in the [stderr] stream even with the most up-to-date changes here so far. Same as before:
> 
> ```
> [stderr] DEBUG:guibot.finder:Next best match is not acceptable
> ```
> 
> Could it be related to Avocado VT doing some additional logging manipulation?

If some messages are in the [stderr] then the test itself send the logs to the stderr, maybe avocado vt or the guibot is doing that. Can you please provide some easy reproducible with guibot for test? Thank you.

> 
> The messages from your test also show in the job log, not sure if this is intended.

When I run `avocado run --store-logging-stream=matplotlib examples/tests/external_logging_stream.py` I can't see any logs in job log can you provide some reproducible as well. Thank you.

Reply:
Since I need to have this logging streams permanently and thus in a config file, does something like this make sense:
```
[root@c4 ~]# cat /etc/avocado/avocado.conf 
[job.run]
store_logging_stream = avocado:DEBUG, aexpect:DEBUG, guibot:DEBUG
```
I cannot find any example in the documentation, example, or blueprints using a config file and the above produces an error that might be caused by me in some way:
```
  File "/mnt/local/intra2net-qa/tp_i2n/tools/cli/run_api/suite.py", line 98, in run_suite
    from avocado.core.app import AvocadoApp
  File "/usr/lib/python3.10/site-packages/avocado/__init__.py", line 36, in <module>
    settings.merge_with_configs()
  File "/usr/lib/python3.10/site-packages/avocado/core/settings.py", line 576, in merge_with_configs
    self.update_option(namespace, value, convert=True)
  File "/usr/lib/python3.10/site-packages/avocado/core/settings.py", line 776, in update_option
    self._namespaces[namespace].set_value(value, convert)
  File "/usr/lib/python3.10/site-packages/avocado/core/settings.py", line 279, in set_value
    self._value = self._as_list(value)
  File "/usr/lib/python3.10/site-packages/avocado/core/settings.py", line 226, in _as_list
    return ast.literal_eval(value)
  File "/usr/lib64/python3.10/ast.py", line 62, in literal_eval
    node_or_string = parse(node_or_string.lstrip(" \t"), mode='eval')
  File "/usr/lib64/python3.10/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    avocado:DEBUG, aexpect:DEBUG, guibot:DEBUG, vncdotool:DEBUG, pyi2ncommon:DEBUG
           ^
SyntaxError: invalid syntax
```

Reply:
> If some messages are in the [stderr] then the test itself send the logs to the stderr, maybe avocado vt or the guibot is doing that. Can you please provide some easy reproducible with guibot for test? Thank you.

Since this requires additional pip-installations on your side let's first see if you have any problems with Avocado VT so that we can exclude the elephant in the room.